File: Backend/MyApp.UserService/DTOs/UserResponse.cs
using MyApp.UserService.Models;

namespace MyApp.UserService.DTOs;

/// <summary>User profile</summary>
/// <param name="Id">Unique identifier</param>
/// <param name="Username">Username</param>
/// <param name="FirstName">First name</param>
/// <param name="LastName">Last name</param>
/// <param name="Email">Email address</param>
/// <param name="TelephoneNumber">Telephone number</param>
/// <param name="Role">Role: 0 — Client, 1 — Employee</param>
/// <param name="IsBlocked">Whether the user is blocked</param>
public record UserResponse(
    int Id,
    string Username,
    string FirstName,
    string LastName,
    string Email,
    string TelephoneNumber,
    Role Role,
    bool IsBlocked
);

File: Mobile/components/user/src/main/java/nekit/corporation/user/data/model/UserResponse.kt
package nekit.corporation.user.data.model

import kotlinx.serialization.Serializable

@Serializable
data class UserResponse(
    val id: Int,
    val username: String,
    val firstName: String,
    val lastName: String,
    val email: String,
    val telephoneNumber: String,
    val role: UserRole,
    val isBlocked: Boolean
)
